legistar/bills.py

```diff
@@ -383,8 +383,11 @@ def text(self, matter_id, latest_version_value=None):
     def legislation_detail_url(self, matter_id):
         gateway_url = self.BASE_WEB_URL + '/gateway.aspx?m=l&id={0}'
 
+        # we want to supress any session level params for this head request
+        # since they could lead to an additonal level of redirect
         legislation_detail_route = self.head(
-            gateway_url.format(matter_id)).headers['Location']
+            gateway_url.format(matter_id),
+            params={k: None for k in self.params}).headers['Location']
```

Comment:
is there a reason you can't do an empty dict here?

Reply:
yes. from http://docs.python-requests.org/en/master/user/advanced/

> Remove a Value From a Dict Parameter

> Sometimes you’ll want to omit session-level keys from a dict parameter. To do this, you simply set that key’s value to None in the method-level parameter. It will automatically be omitted.

Reply:
cool. this strikes me as a little unorthodox. in the spirit of remembering why we made a decision, could you add a comment to the effect of: "Omit session-level keys in the manner recommended by the requests lib" with a link to the docs?

Reply:
cool